### PR TITLE
Nuke Persistent Lock

### PR DESCRIPTION
**Pull request by gmaretic**
**Goals (and why)**:
Simplify code, stop failing on specific legacy sweep use cases due to leaking the lock

**Implementation Description (bullets)**:
Scorched earth policy

**Testing (What was existing testing like?  What have you done to improve it?)**:
Existing tests, I guess

**Concerns (what feedback would you like?)**:
Does someone still rely on this? Even if so, they will have noticed the endpoint disappearing so should be super low risk. Now there is no deduplication for multiple parallel legacy sweep processes -- do we care (no)?

**Where should we start reviewing?**:
idk

**Priority (whenever / two weeks / yesterday)**:
Today

### CODE REVIEW

File: docs/source/cluster_management/backup-restore.rst [hunk 1]
```diff
@@ -1,106 +0,0 @@
-.. _backup-restore:
-
-==================
-Backup and Restore
-==================
-
-Overview
-========
-
-Backup and restore for AtlasDB are non-trivial operations that are not satisfied by simply backing up and restoring the underyling key value store you are using to store your data in.  This page describes how to take advantage of built in AtlasDB utilities, namely the Atlas CLI, in order to perform a live consistent point in time backup and subsequent restore of that backup from total data loss.
```
Comment (jeremyk-91) [inline docs/source/cluster_management/backup-restore.rst:10]: should probably have a follow up to actually preserve some of the information on how one would perform a backup.

**Review by jeremyk-91 — COMMENTED** [review]
Thanks for doing this! Some notes:

- We will need to patch large internal product appropriately, I'd suggest we check with the maintainers if this is safe to remove completely. I can't see a reason why it _wouldn't_ be safe, but we should check.
- There really should exist some documentation on the theory of backup and restore, but that's outside the scope of this change, and it's definitely a positive to eliminate the current documentation, which is _actively wrong_ and may cause *SEVERE DATA CORRUPTION* (TM).

**Review by jeremyk-91 — APPROVED** [review]
we've given a day's notice, and can add this back later on if desired. 👍

**Comment by gmaretic** [comment]
👍

**Pull request merged**